### ISSUE TITLE
Duplicated transducer `init` and `result` functions

### ISSUE THREAD

**Issue by paldepind** [ramda/ramda#1020]
This has been mentioned elsewhere, but there doesn't seem to be an open issue about it.

The plain reducer init function is apparently used 13 places in Ramda:

```
grep -r "return this.xf\['@@transducer/init']()"
_xtake.js:    return this.xf['@@transducer/init']();
_xall.js:    return this.xf['@@transducer/init']();
_xmap.js:    return this.xf['@@transducer/init']();
_xfind.js:    return this.xf['@@transducer/init']();
_xfilter.js:    return this.xf['@@transducer/init']();
_xgroupBy.js:    return this.xf['@@transducer/init']();
_xtakeWhile.js:    return this.xf['@@transducer/init']();
_xdrop.js:    return this.xf['@@transducer/init']();
_xany.js:    return this.xf['@@transducer/init']();
_xdropWhile.js:    return this.xf['@@transducer/init']();
_xfindLastIndex.js:    return this.xf['@@transducer/init']();
_xfindLast.js:    return this.xf['@@transducer/init']();
_xfindIndex.js:    return this.xf['@@transducer/init']();
```

Maybe default `init` and `result` functions should be defined in a helper file to avoid the repetition?

**Comment by buzzdecafe** [comment]
absolutely yes.

While we're at it, there are a number of redundant `@@transducer/result` functions as well:

https://github.com/ramda/ramda/search?utf8=%E2%9C%93&q=return+this.xf%5B%27%40%40transducer%2Fresult%27%5D%28result%5C%29%3B&type=Code

**Comment by paldepind** [comment]
But, where should the helper functions be located? The current scheme seems to be one file in `src/internal/` per helper function.

But separating these two functions across two files might be overkill?

**Comment by buzzdecafe** [comment]
i think something like this would be fine:

```javascript
// file: src/internal/_xfBase.js
module.exports = {
   init: function() { return this.xf['@@transducer/init'](); },
   result: function(result) { return this.xf['@@transducer/result'](result); }
};
```

then:

```
var _xfBase = require('./_xfBase');

// .... :
XThing.prototype.init = _xfBase.init
XThing.prototype.result = _xfBase.result
// etc.
```

**Comment by paldepind** [comment]
Yes. I agree. I'll create a pull request if you don't mind.

**Comment by buzzdecafe** [comment]
sounds good, thank you

**Comment by buzzdecafe** [comment]
fixed in #1022
